server/src/test/java/org/elasticsearch/cluster/routing/allocation/shards/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -974,15 +1040,23 @@ public void testGetRestoreFromSnapshotAffectedResources() {
         for (IndexMetadata indexMetadata : indexMetadataList) {
             indexMetadataMap.put(indexMetadata.getIndex().getName(), indexMetadata);
         }
-        metadataBuilder.indices(indexMetadataMap);
-        metadataBuilder.put(newInstance(featureDataStreamName, List.of(backingIndex.getIndex())));
-        Metadata metadata = metadataBuilder.build();
+
+        Metadata metadata = metadataBuilder.put(
+            ProjectMetadata.builder(projectId)
+                .indices(indexMetadataMap)
+                .put(newInstance(featureDataStreamName, List.of(backingIndex.getIndex())))
+        ).build();
         {
             List<Diagnosis.Resource> affectedResources = ShardAllocationStatus.getRestoreFromSnapshotAffectedResources(
                 metadata,
                 getSystemIndices(featureDataStreamName, ".test-ds-*", ".feature-*"),
-                Set.of(backingIndex.getIndex().getName(), ".feature-index", "regular-index"),
-                10
+                Set.of(
+                    new ProjectIndexName(projectId, backingIndex.getIndex().getName()),
+                    new ProjectIndexName(projectId, ".feature-index"),
+                    new ProjectIndexName(projectId, "regular-index")
+                ),
+                10,
+                projectResolver.supportsMultipleProjects()
```

Comment:
We can just put `false` here instead of using the default project resolver, right?